### PR TITLE
support dataChangeLastModifiedBy display after edited in a namespace

### PR DESCRIPTION
**Pull request by JaredTan95**
support dataChangeLastModifiedBy display after editing in a namespace, not only in the edit history.

image

### CODE REVIEW

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/entity/bo/ItemBO.java [hunk 1]
```diff
@@ -1,8 +1,9 @@
 package com.ctrip.framework.apollo.portal.entity.bo;
 
+import com.ctrip.framework.apollo.common.dto.BaseDTO;
 import com.ctrip.framework.apollo.common.dto.ItemDTO;
 
-public  class ItemBO {
+public  class ItemBO extends BaseDTO {
```
Comment (nobodyiam) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/entity/bo/ItemBO.java]: Looks like this is unnecessary?

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java [hunk 2]
```diff
@@ -224,6 +223,7 @@ private NamespaceBO transformNamespace2BO(Env env, NamespaceDTO namespace) {
     //latest Release
     ReleaseDTO latestRelease;
     Map<String, String> releaseItems = new HashMap<>();
+    Map<String, ItemDTO> deletedItemDTOS = new HashMap<>();
```
Comment (nobodyiam) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java]: how about deletedItemDTOS => deletedItemDTOs ?

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/entity/bo/ItemBO.java (hunk 1 above)
Comment (JaredTan95) [reply apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/entity/bo/ItemBO.java]: Yeah, I will revert it.

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java (hunk 2 above)
Comment (JaredTan95) [reply apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java]: Got it.

**Comment by JaredTan95** [comment]
Do I need to fix `codecov/patch`?

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java [hunk 3]
```diff
@@ -292,6 +296,9 @@ private void fillAppNamespaceProperties(NamespaceBO namespace) {
 
         deletedItem.setDeleted(true);
         ItemDTO deletedItemDto = new ItemDTO();
+        if (deletedItemDTOs.containsKey(key)) {
+          deletedItemDto = deletedItemDTOs.get(key);
+        }
```
Comment (kezhenxu94) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java]: Combine these lines into

```java
ItemDTO deletedItemDto = deletedItemDTOs.computeIfAbsent(key, k -> new ItemDTO());
```

to reduce unnecessary object creation (`ItemDTO`) every time

File: apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/utils/ConfigChangeContentBuilder.java [hunk 4]
```diff
@@ -84,4 +82,19 @@ Item cloneItem(Item source) {
     return target;
   }
 
+  public ConfigChangeContentBuilder convertJsonString(String content) {
+    return gson.fromJson(content, this.getClass());
+  }
```
Comment (kezhenxu94) [inline apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/utils/ConfigChangeContentBuilder.java]: could be static and don't need to create instances when invoke

File: apollo-adminservice/src/main/java/com/ctrip/framework/apollo/adminservice/controller/ItemController.java [hunk 5]
```diff
@@ -135,6 +138,21 @@ public void delete(@PathVariable("itemId") long itemId, @RequestParam String ope
...
   }
 
+  @GetMapping("/apps/{appId}/clusters/{clusterName}/namespaces/{namespaceName}/items/deleted")
+  public List<ItemDTO> findDeletedItems(@PathVariable("appId") String appId,
+                                        @PathVariable("clusterName") String clusterName,
+                                        @PathVariable("namespaceName") String namespaceName) {
+    List<Commit> commits = commitService.find(appId, clusterName, namespaceName, null);
+    if (Objects.nonNull(commits)) {
+      List<Item> deletedItems = commits.stream()
+          .map(item -> new ConfigChangeContentBuilder().convertJsonString(item.getChangeSets()).getDeleteItems())
```
Comment (kezhenxu94) [inline apollo-adminservice/src/main/java/com/ctrip/framework/apollo/adminservice/controller/ItemController.java]: after making `ConfigChangeContentBuilder.convertJsonString` `static`, you don't need `new ConfigChangeContentBuilder()` to create so many instances

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java [hunk 6]
```diff
@@ -244,7 +244,11 @@ private NamespaceBO transformNamespace2BO(Env env, NamespaceDTO namespace) {
     }
 
     //deleted items
-    List<ItemBO> deletedItems = parseDeletedItems(items, releaseItems);
+    itemService.findDeletedItems(appId, env, clusterName, namespaceName).forEach(item->{
```
Comment (kezhenxu94) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/NamespaceService.java]:
```suggestion
    itemService.findDeletedItems(appId, env, clusterName, namespaceName).forEach(item -> {
```

File: apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/utils/ConfigChangeContentBuilder.java (hunk 4 above)
Comment (JaredTan95) [reply apollo-biz/src/main/java/com/ctrip/framework/apollo/biz/utils/ConfigChangeContentBuilder.java]: good catch.

**Pull request merged**